packages/jaeger-ui-components/src/common/TraceName.tsx

```diff
@@ -14,18 +14,24 @@
 
 import * as React from 'react';
 import { css } from 'emotion';
+import cx from 'classnames';
 
 import BreakableText from './BreakableText';
 import LoadingIndicator from './LoadingIndicator';
 import { fetchedState, FALLBACK_TRACE_NAME } from '../constants';
 
 import { FetchedState, TNil } from '../types';
 import { ApiError } from '../types/api-error';
-import { createStyle } from '../Theme';
+import { createStyle, safeSize, Theme, useTheme } from '../Theme';
 
-const getStyles = createStyle(() => {
+const getStyles = createStyle((theme: Theme) => {
   return {
+    TraceName: css`
+      label: TraceName;
+      font-size: ${safeSize(theme.components?.TraceName?.fontSize, 'unset')};
```

Comment:
Should it really be permitted to not set a font size in the first place?
I understand the desire for flexibility, on the other hand - if the font size override gets removed at some point in the future, then I think this should simply be removed as well instead of keeping code around unnecessarily.

Which would mean: Strict type for theme, no need for optional chaining and fallback and maybe renaming the function to something like `toCSSUnitsString`. And in fact - unless that's used multiple times in the code (does it maybe already exist in the wider grafana code base?), I'd do it inline instead of creating a function for it.

Of course these things are also partly up to taste. :)

Reply:
The issue is I still try to treat the jaeger-ui codebase as a separate thing. Will see how that goes later on but the plan was to include it as a separate library that would be used by both grafana and jaeger ui. That is why this done in a more generic way than needed based on the current usage. Good point about the naming though that probably makes more sense.